### PR TITLE
Define a default provider for python3-foo

### PR DESCRIPTION
**Pull request by bnavigator**
My attempt at #69

### CODE REVIEW

File: compile-macros.sh [hunk 1]
```diff
@@ -5,7 +5,8 @@ FLAVORS="python2 python3 python36 python38 pypy3"
 # order of BUILDSET is important, it is copied to order of %pythons,
 # and that determines the last installed binary
 BUILDSET="python2 python3 python36 python38"
-
+DEFAULT_PROVIDER=${BUILDSET##* }
```
Comment (mcepl) [inline compile-macros.sh:8]: Thank you for this pull request, but I don’t like this one. There should be just one source of wisdom on what’s the current provider, and that should be ``%primary_interpreter == 1`` in the appropriate ``python*.spec`` file. We cannot read that macro, but we can find out what ``/usr/bin/python3`` points to, and that should be the provider for ``python3-*`` packages. Not sure how to expand links in Lua (or with [the RPM Lua macros](https://rpm.org/user_doc/lua.html), could ``stat(2)`` help?).

**Comment by bnavigator** [comment]
How about the latest commit? I went with a `%#FLAVOR#_provides` macro that expands some shell commands at runtime and checks for `/usr/bin/pyhton?`.

**Comment by mcepl** [comment]
Looking at https://bugzilla.suse.com/show_bug.cgi?id=1177477 and thinking “How do we deal with update-alternatives?”. Will we automatically switch all alternatives to **default** (not changing already non-default values) when changing the primary Python provider?

**Comment by bnavigator** [comment]
Sorry, I do not quite understand. How is this relevant to the primary provider during buildtime?

- `/usr/bin/python3` is currently not under u-a control.
- This PR does not affect the current u-a priorities.

Do we need to increase the priority of the `%python_install_alternative` macro for the primary provider flavor?

**Comment by bnavigator** [comment]
> Looking at https://bugzilla.suse.com/show_bug.cgi?id=1177477

I do not understand how that situation can occur with the current python36 and python38 packages in Tumbleweed.

**Comment by mcepl** [comment]
> > Looking at https://bugzilla.suse.com/show_bug.cgi?id=1177477
> 
> I do not understand how that situation can occur with the current python36 and python38 packages in Tumbleweed.

I don’t understand either, but I haven’t switched any system from python3.6 to python3.8 yet.

**Comment by bnavigator** [comment]
Sidenote: I noticed a few days ago that Ubuntu has sitelib in python3 rather than python3.X so all flavors use the same sitelib and populate the cache with files for their version.

**Comment by scarabeusiv** [comment]
Yep we won't be changing to Debian layout. This layout we have is used among all rpm distros and it would not help us if users could not inject rpms from Fedora if they really need so.

Debian world has a common folder for the python and specify just minimal supported and thats it.

**Comment by bnavigator** [comment]
It was just a sidenote ;)

**Comment by mcepl** [comment]
OK, please, rebase this PR, and I will happily merge it. Thank you!

**Comment by bnavigator** [comment]
Done.

**Pull request merged**